_source/_posts/2022-02-22-manage-dotnet-microservices.md

```diff
@@ -0,0 +1,461 @@
+- "Learning how to manage multiple microservices? Follow this step-by-step tutorial to build a skeleton of an API gateway and two .NET microservices"
+- "Wondering how to manage to your #dotnet #microservices with #API federation? We've got you covered with this post!"
+image: blog/featured/okta-dotnet-books-quarter.jpg
+type: conversion
+---
+
+For large enterprise companies, boundaries around each team are crucial for maintaining systems that are owned by software engineers. This is accomplished by allowing individual teams to own their systems and expose them via APIs.
+
+You also need to keep certain types of cross-cutting concerns centralized, like select security controls, logging, and routing. If you don't, every API across different teams has to implement common controls like rate-limiting, logging, and authentication. This can lead to maintainability and consistency issues.
+
+However, the [API Federation](https://engineering.salesforce.com/api-federation-growing-scalable-api-landscapes-a0f1f0dad506) pattern keeps cross-cutting concerns located in one centralized lightweight API gateway. For example, by having a centralized place where rate-limiting is processed, and HTTP headers are returned for your company's APIs, each API is kept consistent from the perspective of your clients.
+
+The gateway also short-circuits requests that otherwise might use unnecessary resources. Protecting internal APIs and microservices from [DDoS detection](https://www.kentik.com/kentipedia/ddos-detection/) can be added in one place, which safeguards backend services.
+
+![API gateway](https://i.imgur.com/px7XhgW.png)
+
+In this article, you'll learn how to build [.NET microservices](https://dotnet.microsoft.com/en-us/learn/aspnet/microservices-architecture) that are federated via another .NET API gateway. You'll see how using an API gateway can protect internal services from the outside world by allowing the gateway to handle authentication using [Okta's](https://www.okta.com/) authentication services. These are easily integrated into our .NET microservice.
+
+## What Is API federation?
+
+*APIs* are foundational to modern software development. They enable distributed systems to communicate with each other in a way that allows different teams to build their own systems autonomously.
+
+Larger organizations need to be able to keep cross-cutting and security concerns at a higher-level abstraction so that individual API teams don't need to worry about them.
+
+API federation addresses this need by centralizing cross-cutting and security concerns into a gateway. All incoming API traffic can have security controls applied such as ([DDoS protection](https://www.cloudflare.com/learning/ddos/ddos-mitigation/), [SSL termination](https://avinetworks.com/glossary/ssl-termination/), and [rate-limiting](https://en.wikipedia.org/wiki/Rate_limiting), etc.), and will proxy those requests to internal services.
```

Comment:
```suggestion
API federation addresses this need by centralizing cross-cutting and security concerns into a gateway. All incoming API traffic can have security controls applied such as [DDoS protection](https://www.cloudflare.com/learning/ddos/ddos-mitigation/), [SSL termination](https://avinetworks.com/glossary/ssl-termination/), and [rate-limiting](https://en.wikipedia.org/wiki/Rate_limiting), and will proxy those requests to internal services.
```
remove extra incorrect parentheses, and you don't need etc, since it already says "such as"